code/modules/mob/living/carbon/xenomorph/Xenomorph.dm

```diff
@@ -1138,3 +1141,12 @@
 	if(new_player.mind)
 		new_player.mind_initialize()
 		new_player.mind.transfer_to(target, TRUE)
+
+/mob/living/carbon/xenomorph/proc/roar_and_delay()
+	var/current_time = world.time
+	if(current_time - last_roar_time < 1 SECONDS)
+		to_chat(src, SPAN_WARNING("You must wait before roaring again."))
+		return FALSE
+
+	last_roar_time = current_time
+	playsound(loc, "alien_roar_larva", 15)
```

Comment:
this should be in the emote code itself where you added the call to this proc and place the var where its needed instead of all xenomorphs getting unused var